python_ta/__init__.py

```diff
@@ -37,52 +38,135 @@
     print('You need Python 3.4 or later to run this script')
 
 
-def check_errors(module_name='', reporter=ColorReporter, number_of_messages=5,
-                 config=''):
-    """Check a module for errors, printing a report.
-    """
-    _check(module_name=module_name, reporter=reporter, number_of_messages=number_of_messages, level='error', local_config=config)
+def check_errors(module_name='', config=''):
+    """Check a module for errors, printing a report."""
+    _check(module_name=module_name, level='error', local_config=config)
 
 
-def check_all(module_name='', reporter=ColorReporter, number_of_messages=5,
-              config=''):
+def check_all(module_name='', config=''):
     """Check a module for errors and style warnings, printing a report."""
-    _check(module_name, reporter, number_of_messages, level='all',
-           local_config=config)
+    _check(module_name=module_name, level='all', local_config=config)
+
+
+def _find_pylintrc_same_locale(curr_dir):
+    """Search for a `.pylintrc` configuration file provided in same (user) 
+    location as the source file to check.
+    Return absolute path to the file, or None.
+    `curr_dir` is an absolute path to a directory, containing a file to check.
+    For more info see, pylint.config.find_pylintrc
+    """
+    found_pylintrc_location = None
+    if os.path.exists(os.path.join(curr_dir, '.pylintrc')):
+        found_pylintrc_location = os.path.join(curr_dir, '.pylintrc')
+    elif os.path.exists(os.path.join(curr_dir, 'pylintrc')):
+        found_pylintrc_location = os.path.join(curr_path, 'pylintrc')
+    return found_pylintrc_location
+    
+
+def _apply_nearest_config(linter=None, file_abs_path=''):
```

Comment:
This is better but still too complicated. This helper should be merged with `init_linter`. The merged method should do the following:

0. It should *not* take in a `linter`.
1. Given the `file_abs_path`, determine which configuration to use:
    - If the user's `config` argument is a string, use the config found at that location.
    - If the user's `config` argument is a dictionary, use those options in addition to either the pylintrc file found at the directory of the file being linted, or the default one.
    - If the user's `config` argument is empty, use the pylintrc file found at the directory of the file being linted, or the default one.
2. Return a new `linter` object.

More specifically, do not ever try to re-use a linter object.